### ISSUE TITLE
`coeffs2` does not respect requested size when function is zero

### ISSUE THREAD

**Issue by danfortunato** [chebfun/chebfun#1984]
The second and third arguments of `coeffs2` are ignored if the given function is the zero function. For example,

```
>> f = chebfun2( @(x,y) 0 );
>> coeffs2( f, 3, 3 )

ans =

     0
```

Compare this to
```
>> f = chebfun2( @(x,y) 1 );
>> coeffs2( f, 3, 3 )

ans =

     1     0     0
     0     0     0
     0     0     0
```

**Comment by danfortunato** [comment]
It looks like the fix for this exists in pull request #1874.

**Comment by bhashemi** [comment]
This is fixed with 897461f .
